sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/InternalColumnarRddConverter.scala

```diff
@@ -89,181 +97,330 @@ private object GpuExternalRowToColumnConverter {
-    override def append(
-        row: Row,
-        column: Int,
-        builder: ai.rapids.cudf.HostColumnVector.ColumnBuilder): Unit =
+  private object LongConverter extends TypeConverter {
+    override def append(row: Row,
+      column: Int,
+      builder: ai.rapids.cudf.HostColumnVector.ColumnBuilder): Double = {
       if (row.isNullAt(column)) {
         builder.appendNull()
       } else {
         NotNullLongConverter.append(row, column, builder)
       }
+      8 + VALIDITY
+    }
+
+    override def getNullSize: Double = 8 + VALIDITY
   }
 
-  private object NotNullLongConverter extends FixedWidthTypeConverter {
-    override def append(
-        row: Row,
-        column: Int,
-        builder: ai.rapids.cudf.HostColumnVector.ColumnBuilder): Unit =
+  private object NotNullLongConverter extends TypeConverter {
+    override def append(row: Row,
+      column: Int,
+      builder: ai.rapids.cudf.HostColumnVector.ColumnBuilder): Double = {
       builder.append(row.getLong(column))
+      8
+    }
+
+    override def getNullSize: Double = 8 + VALIDITY
   }
 
-  private object DoubleConverter extends FixedWidthTypeConverter {
-    override def append(
-        row: Row,
-        column: Int,
-        builder: ai.rapids.cudf.HostColumnVector.ColumnBuilder): Unit =
+  private object DoubleConverter extends TypeConverter {
+    override def append(row: Row,
+      column: Int,
+      builder: ai.rapids.cudf.HostColumnVector.ColumnBuilder): Double = {
       if (row.isNullAt(column)) {
         builder.appendNull()
       } else {
         NotNullDoubleConverter.append(row, column, builder)
       }
+      8 + VALIDITY
+    }
+
+    override def getNullSize: Double = 8 + VALIDITY
   }
 
-  private object NotNullDoubleConverter extends FixedWidthTypeConverter {
-    override def append(
-        row: Row,
-        column: Int,
-        builder: ai.rapids.cudf.HostColumnVector.ColumnBuilder): Unit =
+  private object NotNullDoubleConverter extends TypeConverter {
+    override def append(row: Row,
+      column: Int,
+      builder: ai.rapids.cudf.HostColumnVector.ColumnBuilder): Double = {
       builder.append(row.getDouble(column))
+      8
+    }
+
+    override def getNullSize: Double = 8 + VALIDITY
   }
 
-  private object StringConverter extends FixedWidthTypeConverter {
-    override def append(
-        row: Row,
-        column: Int,
-        builder: ai.rapids.cudf.HostColumnVector.ColumnBuilder): Unit =
+  private object StringConverter extends TypeConverter {
+    override def append(row: Row,
+      column: Int, builder: ai.rapids.cudf.HostColumnVector.ColumnBuilder): Double =
       if (row.isNullAt(column)) {
         builder.appendNull()
+        VALIDITY_N_OFFSET
       } else {
-        NotNullStringConverter.append(row, column, builder)
+        NotNullStringConverter.append(row, column, builder) + VALIDITY
       }
+
+    override def getNullSize: Double = OFFSET + VALIDITY
```

Comment:
Nit: There's already a computed constant for this, i.e.: `OFFSET_N_VALIDITY`.  This comment applies to all the places where `OFFSET + VALIDITY` appears.